proto/regen/ecocredit/v1/events.proto

```diff
@@ -36,30 +30,36 @@ message EventCreateBatch {
   string issuer = 2;
 }
 
-// EventCreateBatch is an event emitted when a credit batch is created.
-message EventMintBatchCredits {
+// EventMint is an event emitted when a credits are minted to a credit batch.
+message EventMint {
 
   // batch_denom is the unique identifier of the credit batch within which the
   // credits were minted.
   string batch_denom = 1;
 
-  // A reference to a transaction or an event referencing the transaction
-  // which caused the transfer from other chain or registry.
-  OriginTx origin_tx = 2;
+  // issuer is the account address of the issuer of the credit batch that has
+  // minted credits to the credit batch.
+  string issuer = 2;
```

Comment:
It's not so much about the assertion here as it is about recording who the issuer was. When creating or updating a class or project, we know it was the admin. When issuing batches and minting credits, it could be one of many issuers listed in the credit class. Added here for the same reason I did not remove in `CreateBatch`. This is also why we include things like the `jurisdiction` in `EventRetire`, which is never actually stored on chain.

Reply:
It's not so much about the assertion here as it is about recording who the issuer was. When creating or updating a class or project, we know it was the admin. When issuing batches and minting credits, it could be one of many issuers listed in the credit class. Added here for the same reason I did not remove in `CreateBatch`. This is also why we include things like the `jurisdiction` in `EventRetire`, which is never actually stored in state.

Reply:
Only the original issuer should be able to mint and it looks like it is implemented correctly. Removed here and also removed in `CreateBatch`. Was thinking something different at the time but not needed. :+1:  https://github.com/regen-network/regen-ledger/pull/1086/commits/67b32320f4613d056c23f1037ad96df513835658